templates/etc/fastd/fastd-blacklist.sh.erb

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+PEER_KEY=$1
+
+if /bin/grep -Fq $PEER_KEY /etc/fastd/<%= @mesh_code %>-mesh-vpn/fastd-blacklist.json; then
```

Comment:
On 22/02/15 21:50, Leo Krueger wrote:

> regarding fastdbl: then we should implement this in a nice modular way. I'm not sure at the moment if it works to leave out the fastd key repo, as something like that is not needed in the ffruhr setup if I understand it correctly.

If you use blacklisting, you don't need the fastd key repo anymore; for FFGT I just have a stale snapshot there (we still do collect the keys, more or less to be able to switch back to "only known ppl, thanks").

> json: ok then maybe leave it like that. I thought about working with the json format in the fastd-blacklist.sh file in a nice way. maybe change it to a python script that really parses the format and iterates over the entries. but maybe thats to much effort then...

The blacklist only needs to contain "offending" keys; as this is, IIRC, a synchronous action -- i. d. fastd blocks until the script finishes --, I'd make it as lightweight as possible. If spawning /usr/bin/python is faster that spawning /bin/bash, go for it; for now, I think the shell script is fine as it is (after removing the superflous $1).